auth.js

```diff
@@ -0,0 +1,174 @@
+const loginForm = document.querySelector(".login-form");
+const signupForm = document.querySelector(".signup-form");
+
+const setInputError = (event) => {
+  if (event.target.closest("button")) {
+    return;
+  }
+  if (event.target.value === "") {
+    event.target.nextElementSibling?.remove();
+    event.target.classList.add("input-error");
+    setErrorMessage(event.target, "empty");
+  } else if (event.target.id === "useremail") {
+    event.target.nextElementSibling?.remove();
+    if (event.target.validationMessage) {
+      event.target.classList.add("input-error");
+      setErrorMessage(event.target, "email");
+    } else {
+      event.target.classList.remove("input-error");
+    }
+  } else if (
+    event.target.id === "password" ||
+    event.target.id === "passwordCheck"
+  ) {
+    event.target.nextElementSibling?.remove();
+    if (event.target.value.length < 8) {
+      event.target.classList.add("input-error");
+      setErrorMessage(event.target, "password");
+    } else {
+      event.target.classList.remove("input-error");
+      checkPassword();
+    }
+  } else {
+    event.target.classList.remove("input-error");
+    event.target.nextElementSibling?.remove();
+  }
+};
+
+const checkPassword = () => {
+  const password = document.querySelector("#password");
+  const passwordCheck = document.querySelector("#passwordCheck");
+  if (
+    password.value !== passwordCheck?.value &&
+    passwordCheck?.value.length >= 8
+  ) {
+    passwordCheck.nextElementSibling?.remove();
+    passwordCheck.classList.add("input-error");
+    passwordCheck.classList.remove("input-valid");
+    setErrorMessage(passwordCheck, "passwordCheck");
+  } else if (password.value === passwordCheck?.value) {
+    passwordCheck.nextElementSibling?.remove();
+    passwordCheck.classList.remove("input-error");
+    passwordCheck.classList.add("input-valid");
+  }
+};
+
+const setErrorMessage = (element, type) => {
+  const errorMessage = document.createElement("p");
+  errorMessage.classList.add("error-message", "text-lg", "semibold");
+  if (type === "empty") {
+    if (!element.nextElementSibling) {
```

Comment:
💊 제안
if문의 중첩은 가독성에 좋지 않으니 지금의 경우 아래처럼 하시면 더 좋을 것 같습니다.

```suggestion
  if (element.nextElementSibling) return;  // 이 조건의 해당할 경우 아래 로직이 실행되지 않음
  if (type === "empty") {
```